docs/settings/monitoring-settings.asciidoc

```diff
@@ -70,6 +70,11 @@ both the {es} monitoring cluster and the {es} production cluster.
 +
 If not set, {kib} uses the value of the <<elasticsearch-user-passwd, `elasticsearch.password`>> setting.
 
+`monitoring.ui.elasticsearch.serviceAccountToken`::
+Specifies a {ref}/security-api-create-service-token.html[service account token] to establish a persistent connection in {kib} to the {es} monitoring cluster and to verify licensing status on the {es} monitoring cluster.  This setting is an alternative to `monitoring.ui.elasticsearch.username` and `monitoring.ui.elasticsearch.password`.
++
+If not set, {kib} uses the value of <<elasticsearch-service-account-token, `elasticsearch.serviceAccountToken`>> setting.
+
```

Comment:
@chrisronline @cachedout Maybe you tell me if this sounds okay.  I copied the first paragraph from `monitoring.ui.elasticsearch.username` and `monitoring.ui.elasticsearch.password`, but I'm confused about this part that states:

> Every other request performed by *{stack-monitor-app}* to the monitoring {es}
> cluster uses the authenticated user's credentials, which must be the same on
> both the {es} monitoring cluster and the {es} production cluster.

Is this saying that every other request aside from the previous requests or literally every other request?  And is this case only meant for users who are trying to connect a single Kibana instance to both the monitoring and the production cluster?  Don't we discourage that?

Reply:
This is a bit confusing for sure.

[This comment](https://github.com/elastic/kibana/pull/48437#discussion_r335605865) indicates I had a better understanding of how it worked, but lemme try and recall.

The `.username` credentials are used to authenticate requests against the _monitoring cluster_ (which can be the same as the main/production cluster) for two purposes (just looking at master code to figure this out which consists of searching for `asInternalUser` and making sure it's coming from the custom client and not the one provided by core):

- For the license poller to ensure the _monitoring cluster_ has the right license to enable cluster alerts (this is most likely deprecated as we don't rely on watcher-based cluster alerts anymore)
- To collect [telemetry data](https://github.com/elastic/kibana/blob/main/x-pack/plugins/monitoring/server/kibana_monitoring/collectors/get_usage_collector.ts#L99)

It seems very inconsistent without a good reason.

I think it's most legacy reasons we still require this, and it likely can change but you'll want to do some testing to make sure you've identified any other edge case.

Reply:
Thanks for the info. Think I get it more now.  Would these settings only be used in the case where `monitoring.ui.elasticsearch.hosts` is set to something?  Maybe that's obvious, but I think that's where I was getting confused.

Reply:
Yup, I think that's right. We only use [this config if `hosts` are set](https://github.com/elastic/kibana/blob/main/x-pack/plugins/monitoring/server/es_client/instantiate_client.ts#L39)

Reply:
Is there a particular reason we need to use "asInternalUser" and can't use the  "authenticated user's credentials" instead?  Sounds like probably can just remove the code anyway but will need to verify as you said.  If I understand correctly the "asInternalUser" represents the `monitoring.ui.elasticsearch.username`

Reply:
`asInternalUser` is typically used when things need to happen without an authorized user. We will continue to try and send monitoring metrics through the bulk uploader even if no one ever logs into Kibana.

> correctly the "asInternalUser" represents the monitoring.ui.elasticsearch.username

For Stack Monitoring, yes - but in general, it means whichever user is configured in the `kibana.yml` (same applies for `elasticsearch.username` too)

Reply:
Thanks for all the info!